src/synapse/synapse_handler.rs

```diff
@@ -32,16 +32,24 @@ pub async fn accept_room_invitation<'a>(
     room_message_body: Option<&str>,
     synapse: &SynapseComponent,
 ) -> Result<(), CommonError> {
-    // Check if it's a `accept` event.
-    if room_event != FriendshipEvent::ACCEPT {
-        return Ok(());
+    let joined_rooms = synapse.get_joined_rooms(token).await;
+    match joined_rooms {
+        Ok(rooms) => {
+            if !rooms.joined_rooms.contains(&room_id.to_string()) {
+                // The room exists of a previous interaction between users, but the current user hasn't joined yet
+                let joined_room = synapse
+                    .join_room(token, &room_id, room_event, room_message_body)
+                    .await;
+                match joined_room {
+                    Ok(_) => Ok(()),
+                    Err(err) => Err(err),
+                }
```

Comment:
nit: Can be simplified with a map for the successful branch.
```suggestion
                joined_room.map(|_| ())
```